x/gov/keeper/msg_server.go

```diff
@@ -55,36 +64,62 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *v1beta1.MsgSubmitP
 		sdk.NewEvent(
 			sdk.EventTypeMessage,
 			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
-			sdk.NewAttribute(sdk.AttributeKeySender, msg.GetProposer().String()),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.GetProposer()),
 		),
 	)
 
-	submitEvent := sdk.NewEvent(types.EventTypeSubmitProposal, sdk.NewAttribute(types.AttributeKeyProposalType, msg.GetContent().ProposalType()))
 	if votingStarted {
-		submitEvent = submitEvent.AppendAttributes(
+		submitEvent := sdk.NewEvent(types.EventTypeSubmitProposal,
 			sdk.NewAttribute(types.AttributeKeyVotingPeriodStart, fmt.Sprintf("%d", proposal.ProposalId)),
 		)
+
+		ctx.EventManager().EmitEvent(submitEvent)
 	}
 
-	ctx.EventManager().EmitEvent(submitEvent)
-	return &v1beta1.MsgSubmitProposalResponse{
+	return &v1beta2.MsgSubmitProposalResponse{
 		ProposalId: proposal.ProposalId,
 	}, nil
 }
 
-func (k msgServer) Vote(goCtx context.Context, msg *v1beta1.MsgVote) (*v1beta1.MsgVoteResponse, error) {
+func (k msgServer) ExecLegacyContent(goCtx context.Context, msg *v1beta2.MsgExecLegacyContent) (*v1beta2.MsgExecLegacyContentResponse, error) {
```

Comment:
Do we need to check that the authority address matches that of the governance account?

Reply:
Added